staging/src/k8s.io/component-base/config/testing/helpers.go

```diff
@@ -76,6 +76,10 @@ func matchOutputFile(t *testing.T, actual []byte, expectedFilePath string) {
 		t.Fatalf("couldn't read test data: %v", err)
 	}
 
+	// The YAML encoder is writing \r\n line endings on Windows, which may not match
```

Comment:
I'm really surprised by this, and this seems like something we should fix in the yaml encoder, not work around. There are lots of unit tests where we compare yaml-serialized data to golden testdata files, are those failing as well? (e.g. everything under k8s.io/api)

Reply:
do you have a pointer to where the \r\n endings are being written by the yaml encoder on Windows? I couldn't find that

Reply:
if you run `go test k8s.io/api` on windows, does that pass? that uses the same yaml encoder to serialize data and does byte comparison to os-independent fixtures, so if that's passing, it seems like this test has an os-specific issue we should fix, not mask with normalization

Reply:
I should have looked a bit more. I stopped at this point https://github.com/kubernetes/kubernetes/blob/19ab5b69108172a1c1725a23fdf4e30eba4ff629/vendor/gopkg.in/yaml.v2/emitterc.go#L39, and assumed that it was putting \r\n. It turns out that case can never happen, since ``emitter.line_break`` is never initialized (meaning it's ``yaml_ANY_BREAK``, which defaults to ``yaml_LN_BREAK``: https://github.com/kubernetes/kubernetes/blob/19ab5b69108172a1c1725a23fdf4e30eba4ff629/vendor/gopkg.in/yaml.v2/emitterc.go#L292 . At this point, you can't configure the yaml encoder to write \r\n.

It turns out that those yaml files really were checked out by git with \r\n endings, causing this type of failure:

```
=== RUN   TestDefaultsYAML/CredentialProviderConfig_default_v1beta1
    helpers.go:89: Output does not match expected, diff (- want, + got):
          strings.Join({
                "apiVersion: kubelet.config.k8s.io/v1beta1",
        -       "\r",
                "\nkind: CredentialProviderConfig",
        -       "\r",
                "\nproviders: null",
        -       "\r",
                "\n",
          }, "")

    helpers.go:104: if the diff is expected because of a new type or a new field, re-run with UPDATE_COMPONENTCONFIG_FIXTURE_DATA=true to update the compatibility data or generate missing files
```

This can happen if the ``core.autocrlf`` option is set to true (``git config --global core.autocrlf true``) before cloning the project, for example. On a fresh environment, with this option set to false, this type of error won't occur, the files are checked out with just LF line endings.

I'm ok with removing these changes here, meaning we leave CRLF case unresolved.

Reply:
yeah, I'd rather fix that with something like https://github.com/golang/go/blob/master/.gitattributes instead of in-test fixup